lizmap/modules/lizmap/classes/qgisForm.class.php

```diff
@@ -267,6 +272,43 @@ public function __construct($layer, $form, $featureId, $loginFilteredOverride)
                 }
             }
         }
+
+        // Set form's private data
+        $privateData = array();
+        $privateData['liz_repository'] = $layer->getProject()->getRepository()->getKey();
+        $privateData['liz_project'] = $layer->getProject()->getKey();
+        $privateData['liz_layerId'] = $layer->getId();
+        $privateData['liz_featureId'] = $featureId;
+        $privateData['liz_geometryColumn'] = $this->dbFieldsInfo->geometryColumn;
+
+        $privateData['qgis_controls'] = array();
+        foreach($this->formControls as $fieldName => $formControl) {
+            $privateData['qgis_controls'][$formControl->ref] = array(
+              'fieldName' => $formControl->fieldName,
+              'defaultValue' => $formControl->defaultValue,
+              'fieldEditType' => $formControl->fieldEditType,
+              'fieldAlias' => $formControl->fieldAlias,
+              // 'rendererCategories' => $formControl->rendererCategories, // needs to be parsed
+              'fieldDataType' => $formControl->fieldDataType,
+              'isReadOnly' => $formControl->isReadOnly,
+              'required' => $formControl->required,
+              'valueRelationData' => $formControl->valueRelationData,
+              'relationReferenceData' => $formControl->relationReferenceData,
+              'uniqueValuesData' => $formControl->uniqueValuesData,
+            );
+        }
+
+
+        $attributeEditorForm = $this->getAttributesEditorForm();
+        if ($attributeEditorForm) {
+            $groupVisibilities = $attributeEditorForm->getGroupVisibilityExpressions();
+            $qgis_groupDependencies = qgisExpressionUtils::getCriteriaFromExpressions($groupVisibilities);
```

Comment:
@rldhont `Class 'qgisExpressionUtils' not found	/lizmap_web_client/lizmap/modules/lizmap/classes/qgisForm.class.php`

Reply:
`qgisExpressionUtils` is declared in the autoload section of the `module.xml` file. It has to be autoloaded.

Reply:
How to autoload it ?

Reply:
@rldhont I still have this issue and don't know how to autoload it.